src/components/Tables/Table.tsx

```diff
@@ -126,7 +128,16 @@ const Table = <Data,>({
     columnDisplayFunctions = {},
     columnStyleOptions = {},
     autoFilter = true,
+    selectedCheckboxes,
+    setSelectedCheckboxes,
 }: Props<Data>): React.ReactElement => {
+    const [selectedCheckboxesDefault, setSelectedCheckboxesDefault] = useState(
+        new Array<boolean>(inputData.length).fill(false)
+    );
+
+    selectedCheckboxes = selectedCheckboxes ?? selectedCheckboxesDefault;
+    setSelectedCheckboxes = setSelectedCheckboxes ?? setSelectedCheckboxesDefault;
```

Comment:
I think it's an anti-pattern to override the props. 

Another thing we should consider - do we really want the "default" stuff? It might be better if the table is clear about who should manage the checkbox state instead. What I would've expected is something like below, where we don't even need to pass the `setSelectedRowIndices` to the table. The table then uses `selectedRowIndices` to simply display the checkboxes in the correct states, and lets the parent do whatever it needs do to the checkbox states.

```tsx
// in Parcels page

const [selectedRowIndices, setSelectedRowIndices] = useState<number[]>([])

const selectOrDeselectRow = (rowIndex: number) => {
  setSelectedRowIndices(currentIndices => {
    if (currentIndices.includes(rowIndex)) {
      return currentIndices.filter(index => index !== rowIndex)
    }
    return currentIndices.concat([rowIndex])
  })

  // do whatever else we may want to do
}

return (
  ...
  <Table selectedRowIndices={selectedRowIndices} onCheckboxClicked={selectOrDeselectRow} ... />
)
```

Reply:
Thank you, have implemented this, and similarly for ActionBar + its children